File: 프로그래머스/lv1/72410. 신규 아이디 추천/신규 아이디 추천.py
import re
def solution(new_id):
    new_id = new_id.lower()
    new_id = re.sub('[^a-z0-9\-_.]', '', new_id)
    new_id = re.sub('\.+', '.', new_id)
    new_id = new_id.strip('.')
    if not new_id: new_id = "a"
    if len(new_id) > 15: new_id = new_id[:15].rstrip('.')
    if len(new_id) < 3: new_id += new_id[-1] * (3 - len(new_id))
    return new_id

File: 프로그래머스/lv1/72410. 신규 아이디 추천/신규 아이디 추천.java
class Solution {
    public String solution(String new_id) {
        new_id = new_id.toLowerCase();
        new_id = new_id.replaceAll("[^a-z0-9\\-_.]", "");
        new_id = new_id.replaceAll("\\.+", ".");
        if (!new_id.isEmpty() && new_id.charAt(0) == '.')
            new_id = new_id.substring(1);
        if (!new_id.isEmpty() && new_id.charAt(new_id.length() - 1) == '.')
            new_id = new_id.substring(0, new_id.length() - 1);
        if (new_id.isEmpty())
            new_id = "a";
        if (new_id.length() > 15)
            new_id = new_id.charAt(14) != '.' ? new_id.substring(0, 15) : new_id.substring(0, 14);
        if (new_id.length() < 3)
            new_id += Character.toString(new_id.charAt(new_id.length() - 1)).repeat(3 - new_id.length());
        
        return new_id;
    }
}